src/index.js

```diff
@@ -0,0 +1,172 @@
+import UploaderService from './service';
+
+const KILOBYTE = 1000,
+    MEGABYTE = 1024;
+
+class UploadController {
+
+    constructor({url, attachments = {}, maxFiles = 10, headers = {}, optionsObject = {}, updateCb, onLoad, maxFilesText = ''}) {
+        this.attachments = attachments;
+        this.maxFiles = maxFiles;
+        this.url = url;
+        this.optionsObject = optionsObject;
+        this.headers = headers;
+        this.fileCounter = Object.keys(this.attachments).length;
+        this.updateCb = updateCb;
+        this.onLoad = onLoad;
+        this.maxFilesText = maxFilesText;
+
+        this.onAttach = this.onAttach.bind(this);
+        this.onAbort = this.onAbort.bind(this);
+    }
+
+    static itemId(name) {
+        return `file_${name.match(/[a-zA-Z0-9_]/gi).join('')}`;
+    }
+
+    static maxFilesReached(fileCount, maxFiles) {
+        return fileCount >= maxFiles;
+    }
+
+    static stripNameFromExtension(name) {
```

Comment:
am using it. 
![image](https://cloud.githubusercontent.com/assets/11255103/25130635/91371f6a-244b-11e7-8b5f-f79fafa7581e.png)